### DISCUSSION TITLE
Lighthouse user-flow with puppeteer not working

### DISCUSSION THREAD

**Discussion by kerstin97** [GoogleChrome/lighthouse#14952] [Q&A] [answered]
Hello ✋ 

so it's my first time using lighthouse and puppeteer and I don't understand what I'm doing wrong.
I try to test my website with a userflow ([Lighthouse Userflow](https://github.com/GoogleChrome/lighthouse/blob/main/docs/user-flows.md#creating-a-flow)).

This is my code: [Repository](https://github.com/kerstin97/lighthouse-puppeteer-userflows)

When I run `node remix_test_with_lighthouse.js`, the first navigation of the flow is working. But then I get a timeout on the second `flow.navigate`.  I've tried to disable  the timeout limit with `await page.setDefaultNavigationTimeout(0);` but then it stucks forever. 

Also if I remove the second `flow.navigate`, it gets stuck on the `flow.timespan` method. 

Generally the basic test-code without lighthouse is working (`node remix-test.js`), so it's something with the userflow. 😢 

This is the error(using node v18.15.0):
```
/node_modules/lighthouse/core/gather/driver/wait-for-condition.js:85
      reject(new LighthouseError(LighthouseError.errors.NO_FCP));
             ^

LighthouseError: NO_FCP
    at Timeout.<anonymous> (file:///Users/kerstinreichinger/lighthouse-puppeteer-userflows/node_modules/lighthouse/core/gather/driver/wait-for-condition.js:85:14)
    at listOnTimeout (node:internal/timers:569:17)
    at process.processTimers (node:internal/timers:512:7) {
  code: 'NO_FCP',
  friendlyMessage: 'The page did not paint any content. Please ensure you keep the browser window in the foreground during the load and try again. (NO_FCP)',
  lhrRuntimeError: true
}

Node.js v18.15.0
```

Thank you in advande for helping me,
Kerstin

**Comment by adamraine** [comment, marked as answer]
The `flow.navigate` method is waiting for a page navigation, but it never sees one because the `page.click` command does not actually initiate a real page navigation. The page is doing a "soft navigation" instead which cannot be tested with navigation mode. To audit a soft navigation right now, you need to use Lighthouse timespan mode.

We plan on adding support for auditing soft navigations in the future (#14573), but the Chrome tooling behind it is still experimental. If you're interested in being a trailblazer, you can also check out https://github.com/adamraine/lighthouse-plugin-soft-navigation.

**Reply by kerstin97** [reply]
Thank you so much for answering so quickly. Maybe after my bachelor degree I'll come back to this issue 😄
